runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -1355,95 +1356,49 @@ scan_hex_helper(J9JavaVM *javaVM, char **cursor, UDATA *value, const char *argNa
 }
 
 /**
- * Wrapper for scan_udata_helper, that provides parsing for memory sizes.
- * User should be able to specify the size in GiBs, MiBs, or KiBs (with G,g,M,m,K,k suffixes) or
- * in bytes (no suffix)
+ * Wrapper for scan_udata_memory_size, that provides readable error messages.
+ * @param cursor address of the pointer to the string to parse for the udata.
+ * @param value address of the storage for the udata to be read.
+ * @param argName string containing the argument name to be used in error reporting.
  * @return true if parsing was successful, false otherwise.
  */
 bool
 scan_udata_memory_size_helper(J9JavaVM *javaVM, char **cursor, UDATA *value, const char *argName)
 {
 	PORT_ACCESS_FROM_JAVAVM(javaVM);
+	UDATA result = scan_udata_memory_size(cursor, value);
 
-	if(!scan_udata_helper(javaVM, cursor, value, argName)) {
-		return false;
-	}
-	
-	if(try_scan(cursor, "T") || try_scan(cursor, "t")) {
-		if (0 != *value) {
-#if defined(J9VM_ENV_DATA64)
-			if (*value <= (((UDATA)-1) >> 40)) {
-				*value <<= 40;
-			} else
-#endif /* defined(J9VM_ENV_DATA64) */
-			{
-				j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_GC_OPTIONS_VALUE_OVERFLOWED, argName);
-				return false;
-			}
-		}
-	} else if(try_scan(cursor, "G") || try_scan(cursor, "g")) {
-		if (*value <= (((UDATA)-1) >> 30)) {
-			*value <<= 30;
-		} else {
-			j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_GC_OPTIONS_VALUE_OVERFLOWED, argName);
-			return false;
-		}
-	} else if(try_scan(cursor, "M") || try_scan(cursor, "m")) {
-		if (*value <= (((UDATA)-1) >> 20)) {
-			*value <<= 20;
-		} else {
-			j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_GC_OPTIONS_VALUE_OVERFLOWED, argName);
-			return false;
-		}
-	} else if(try_scan(cursor, "K") || try_scan(cursor, "k")) {
-		if (*value <= (((UDATA)-1) >> 10)) {
-			*value <<= 10;
-		} else {
-			j9nls_printf(PORTLIB, J9NLS_ERROR, J9NLS_GC_OPTIONS_VALUE_OVERFLOWED, argName);
-			return false;
-		}
+	/* Report Errors */
+	if (1 == result) {
```

Comment:
As far as you move functions it is make sense to replace return codes 1 and 2 to defined values at the same time

Reply:
There are broader changes to ensure that these functions use defined values. A large portion of the j9argscan methods already pass OPTION_MALFORMED/OPTION_OVERFLOW, however there are a notable few that return hard-coded values. I need to create another commit to address this, and it will need to include:
1) Replace the returned values of these methods that return positive integers with returning OPTION_MALFORMED/OPTION_OVERFLOW, or any other respective option error defined value.
2) Change the caller of each of these to expect the defined values.

I suggest merging these changes independently so we're able to clean-up the remainder of j9argscan as well.

Reply:
It is ok for me. Would you please create an issue to reflect required changes (if it has not been created before) and mention it here?

Reply:
See https://github.com/eclipse/openj9/issues/10461